tests/components/shelly/test_config_flow.py

```diff
@@ -66,8 +67,36 @@ async def test_form_auth(hass):
             {"host": "1.1.1.1"},
         )
 
-    assert result2["type"] == "abort"
-    assert result2["reason"] == "auth_not_supported"
+    assert result2["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "aioshelly.Device.create",
+        return_value=Mock(
+            shutdown=AsyncMock(),
+            settings={"name": "Test name", "device": {"mac": "test-mac"}},
+        ),
+    ), patch(
+        "homeassistant.components.shelly.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.shelly.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"],
+            {"username": "test username", "password": "test password"},
+        )
+
+    assert result3["type"] == "create_entry"
+    assert result3["title"] == "Test name"
+    assert result3["data"] == {
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6103615)